src/stores/spaces/SpaceStore.ts

```diff
@@ -380,19 +385,31 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
         return false;
     }
 
-    public getSpaceFilteredRoomIds = (space: SpaceKey): Set<string> => {
+    // get all rooms in a space
+    // including descendant spaces
+    public getSpaceFilteredRoomIds = (space: SpaceKey, includeDescendantSpaces = true): Set<string> => {
```

Comment:
there are a lot of call sites with `includeDescendantSpaces = true` and no caching, it feels like that may have detrimental performance effects for users with many deep space graphs.

Reply:
Yeah, I will do some profiling on this. Implementing hierarchical SpaceFilterConditions should remove quite a few of the calls, but there are still quite a few coming from notification state management too.

Reply:
Notification states could be hierarchical also

Reply:
Out of interest was there a reason to not do this or is this a possible future improvement?

Reply:
I have another PR lined up for hierarchical SpaceFilterConditions after this is merged, wanted to limit the scope here. Hierarchical notifications don't directly contribute to my ticket - aggregate room list toggling - so will only be done time permitting.